packages/api/amplify_api/lib/src/graphql/graphql_response_decoder.dart

```diff
@@ -31,19 +31,23 @@ class GraphQLResponseDecoder {
 
   static GraphQLResponseDecoder get instance => _instance;
 
-  GraphQLResponse<T> decode<T>(
-      {required GraphQLRequest request,
-      String? data,
-      List<GraphQLResponseError>? errors}) {
-    if (data == null) {
-      return GraphQLResponse(data: null, errors: errors);
-    }
+  GraphQLResponse<T> decode<T>({
+    required GraphQLRequest request,
+    required Map<String, dynamic> response,
+  }) {
+    final errors = _deserializeGraphQLResponseErrors(response);
+    final data = response['data'];
+
     // If no modelType fallback to default (likely String).
     final modelType = request.modelType;
     if (modelType == null) {
       if (T == String || T == dynamic) {
+        // Preserve `null`. json.encode(null) returns "null" not `null`
+        final encodedData = data != null ? json.encode(data) : null;
```

Comment:
Wondering if it would be better to return a Map instead of a String by default since this is how most other libraries do it and a string is not really useful until you decode it anyway

Reply:
Interesting idea. Agree string not really useful. The response data is always decoded to the `<T>` of the request so it doesn't seem like it would work when explicit `<String>` like is in lots of our docs. Maybe we could do that if `<dynamic>` or `<Map>`?

Reply:
I'm thinking long-term we'll want to get rid of `String` and `dynamic` support. Not sure we're quite ready to do that - maybe as part of the modelgen changes